tests: access kernel object with private data using system call

maksimmasalski opened the pull request:
When defining system calls, it is very important to ensure that
access to the API’s private data is done exclusively through system
call interfaces. Private kernel data should never be made available
to user mode threads directly. For example, the k_queue APIs were
intentionally not made available as they store bookkeeping
information about the queue directly in the queue buffers which are
visible from user mode.

Signed-off-by: Maksim Masalski <maksim.masalski@intel.com>

maksimmasalski commented:
Removed that test from #25641, created separate PR for that test.

maksimmasalski commented:
I still see false-positive CI error for the `mps2_an52`1 and `qemu_x86_nopae`
How to avoid it?

andrewboie commented on tests/kernel/queue/src/test_queue_contexts.c:
```diff
@@ -29,6 +29,16 @@ static K_THREAD_STACK_DEFINE(tstack1, STACK_SIZE);
+
+void k_sys_fatal_error_handler(unsigned int reason, const z_arch_esf_t *pEsf)
+{
+	if (reason != K_ERR_KERNEL_OOPS) {
```
I think in the fullness of time it would be best to add optional fatal error handling directly into Ztest so that various test cases don't all have to do it themselves.

tyrantlizard77 reviewed (approved):
LGTM

tyrantlizard77 reviewed (approved):
LGTM

tyrantlizard77 reviewed (approved):
LGTM

maksimmasalski commented:
Fixed CI, next time if test expected to fail, necessary to add `ignore_faults `tag in .yaml file

Commit:
tests: access kernel object with private data using system call

When defining system calls, it is very important to ensure that
access to the API’s private data is done exclusively through system
call interfaces. Private kernel data should never be made available
to user mode threads directly. For example, the k_queue APIs were
intentionally not made available as they store bookkeeping
information about the queue directly in the queue buffers which are
visible from user mode.

Signed-off-by: Maksim Masalski <maksim.masalski@intel.com>

enjiamai commented on tests/kernel/queue/src/test_queue_contexts.c:
```diff
@@ -398,3 +408,39 @@ void test_multiple_queues(void)
+	/* try to access to private kernel data, will happen kernel oops */
+	k_queue_is_empty(&queue);
+
+	ret = TC_FAIL;
```
Maybe I am wrong. Is "ret = TC_FAIL" here really necessary? I mean, I didn't see a zassert marco to check the result in the code.   
How about put a ztest_test_pass() inside the k_sys_fatal_handler(), and never return?

enjiamai reviewed (commented):
Only one question here, could you please help to clarify? thanks!

LeiW000 commented on tests/kernel/queue/src/test_queue_contexts.c:
```diff
@@ -398,3 +408,39 @@ void test_multiple_queues(void)
+ *   thread doesn't have permission on k_queue object with private kernel data.
+ *
+ * @ingroup kernel_memprotect_tests
+ */
```
Could you re-write the doxygen comment as you did in #29528? We need them for generating test specification.

By the way, please also update the doxygen comments of your other PRs.

maksimmasalski replied:
I don't see that it is in Zephyr Community Guideline https://docs.zephyrproject.org/latest/development_process/documentation.html?highlight=doxygen

LeiW000 replied:
We need it for test specification. Maybe I'm going to propose to use the new format of doxygen comment for all test cases. Of couse, this is another topic. 

For now, It's OK to get the PR approved/merged first.

maksimmasalski commented:
@carlescufi Hello, Carles. Could you merge that PR?
